### DISCUSSION TITLE
在ThreadLocal.h中是否存在该bug？

### DISCUSSION THREAD

**Discussion by chachabules** [chenshuo/muduo#617] [Q&A] [answered]
代码 
 T& value()
  {
    T* perThreadValue = static_cast<T*>(pthread_getspecific(pkey_));
    if (!perThreadValue)
    {
      T* newObj = new T();
      MCHECK(pthread_setspecific(pkey_, newObj));
      perThreadValue = newObj;
    }
    return *perThreadValue;
  }
中”perThreadValue = newObj;”是否错误赋值，是否会导致得到的T值不正确

**Comment by chenshuo** [comment, marked as answer]
赋值有什么错误？
为什么你认为“会导致得到的T值不正确”？

**Reply by chachabules** [reply]
抱歉 ，理解错了这个函数判断分支的作用
